### DISCUSSION TITLE
Make event types extendable

### DISCUSSION THREAD

**Discussion by cgeorgilakis** [keycloak/keycloak#34341] [Ideas]
When using Keycloak extensions, you can not add new login events types with EventBuilder.
Even if this change to support adding String, new event type will not be searchable.
I believe the same stands also for admin events with the difference that resource type is String in the back end.
I believe that Event Type should be extensible in order for Keycloak extensions to be able to have their own events without having to extend Keycloak core code.

**Comment by danielFesenmeyer** [comment]
Would be similar to Admin Events, where #resourceType is a string now, which can be set by #setResourceTypeAsString or #setResourceType (the latter with param of enum ResourceType, providing built-in resource types).
